### DISCUSSION TITLE
2.0.1 with 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by dongri727** [serverpod/serverpod#2410] [Q&A]
I have raised Serverpod to 2.0.1 and deployed with DartSDK 3.4.3 in deployment-aws and now I have a 502BadGateway. Is there any other code that I should fix if I upgrade to 2.0.1?

**Comment by dongri727** [comment]
<img width="510" alt="スクリーンショット 2024-06-24 6 46 49" src="https://github.com/serverpod/serverpod/assets/84567162/f84109a7-5b9e-4146-a99f-cab301c291be">

**Comment by dongri727** [comment]
back to 3.1.5 stay 2.0.1   I couldn't deploy
back to 2.0.0 stay 3.4.3  I could deploy but 502 bad gateway
back to  both 2.0.0 and 3.1.5  I could deploy and website reopened

**Comment by dongri727** [comment]
I'll stay in this condition for a while, but I also want to enjoy the benefits of Upgrade.

**Comment by SandPod** [comment]
Hey!

This seems strange, nothing that I can think of should produce a 502 bad gateway request.

This more likely feels like the Serverpod instance might be crashing on startup or getting stuck.

What do the server logs say when you try and run Dart 3.4.3 and Serverpod 2.0.1 on AWS? Could you paste them here?

**Reply by hidayet-poscu** [reply]
aws
[ec2-user@ip-10-0-6-131 ~]$ dart --version
Dart SDK version: 3.5.0 (stable) (Tue Jul 30 02:17:59 2024 -0700) on "linux_x64"
[ec2-user@ip-10-0-6-131 ~]$

vscode
hidayet@192 terraform % dart --version
Dart SDK version: 3.5.0 (stable) (Tue Jul 30 02:17:59 2024 -0700) on "macos_arm64"
Flutter 3.24.0
serverpod: 2.0.2

conclusion: 502 Bad Gateway

please help

**Reply by SandPod** [reply]
Could you share the logs for your running server instance?

**Reply by hidayet-poscu** [reply]
> Çalışan sunucu örneğinizin günlüklerini paylaşabilir misiniz?

---
"I deleted and closed the account. My guess is that there is an issue with Serverpod related to the latest AWS policies. Previously, those who set up the environment were able to get it working by updating the Dart version. However, for new accounts like mine, no matter what you do, it doesn’t work."

---

**Reply by hidayet-poscu** [reply]
> Çalışan sunucu örneğinizin günlüklerini paylaşabilir misiniz?

The connection with Google Cloud was successful. The issue is with AWS.

The following section on the documentation needs to be updated: [Deploy Your Serverpod Code](https://docs.serverpod.dev/deployments/deploying-to-gce-terraform#deploy-your-serverpod-code).

Finally, click on "Add additional owners" and add the email from the service account you created earlier. Doing this will allow Terraform to make any required changes to your SSL certificates. After visiting [Google Search Console](https://search.google.com/), you need to add the user under "Settings > Users and Permissions" and ensure they have owner permissions. 
example:
Add the service email: eslimasoftware19@eslimasoftwareograsist.iam.gserviceaccount.com as an owner.
"I would appreciate it if you could let me know once the issue with AWS has been resolved."

**Reply by Isakdl** [reply]
@dongri727 We have found some issues and updated both the templates and the docs!

See here how to upgrade the Dart version for your deployment, https://docs.serverpod.dev/next/deployments/deploying-to-aws#configure-dart-version

I hope this may resolve your issue.
